virtcontainers/fc.go

```diff
@@ -737,30 +739,33 @@ func (fc *firecracker) resumeSandbox() error {
 	return nil
 }
 
-func (fc *firecracker) fcAddVsock(vs kataVSOCK) error {
-	span, _ := fc.trace("fcAddVsock")
+func (fc *firecracker) fcAddVsock(hvs types.HybridVSock) error {
+	span, _ := fc.trace("fcAddHybridVsock")
 	defer span.Finish()
 
+	udsPath := hvs.UdsPath
+	if fc.jailed {
+		udsPath = filepath.Join("/", defaultHybridVSocketName)
+	}
+
 	vsockParams := ops.NewPutGuestVsockByIDParams()
 	vsockID := "root"
-	ctxID := int64(vs.contextID)
-	udsPath := ""
+	ctxID := int64(3)
 	vsock := &models.Vsock{
 		GuestCid: &ctxID,
-		UdsPath: &udsPath,
-		VsockID: &vsockID,
+		UdsPath:  &udsPath,
+		VsockID:  &vsockID,
 	}
 	vsockParams.SetID(vsockID)
 	vsockParams.SetBody(vsock)
-	_, err := fc.client().Operations.PutGuestVsockByID(vsockParams)
+
+	e, err := fc.client().Operations.PutGuestVsockByID(vsockParams)
 	if err != nil {
-		return err
+		return fmt.Errorf("%v: %v", e.Error(), err)
```

Comment:
Was this just for debugging? e.Error() should not be required as it simply returns the error string.

Reply:
ouch! good catch , fixing right away